man/man5/zpool-features.5

```diff
@@ -1058,8 +1056,6 @@ on \fBzstd\fR compression of any dataset by running
 This feature becomes \fBactive\fR once a \fBcompress\fR property has been set to
 \fBzstd\fR, and will return to being \fBenabled\fR once all filesystems that
 have ever had their compress property set to \fBzstd\fR are destroyed.
```

Comment:
I propose to just add a comment about grub here, like
```suggestion
have ever had their compress property set to \fBzstd\fR are destroyed.

Booting off of \fBzstd\fR-compressed root pools via GRUB is not yet supported.
```

Reply:
Trivially false as well. I have machines that for [reasons not relevant here] I cannot EFI-boot, and they use /boot on a traditional filesystem, normal ZFS rootfses, all loaded off GRUB2.

GRUB2's feature-set (and limiting a pool thereto) is already out-laid at the top of this page, and which subtrees of the hierarchy exist on which pools (if any) is an administrative decision (and how that relates to any boot-loader doubly so), and entirely irrelevant to this listing of pool features. This is a confusing, rotting appendix of OpenSolaris/Illumos, same way as restricting bootfs= was.

Reply:
> and they use /boot on a traditional filesystem

So GRUB boots from `/boot on a traditional filesystem` in this case, not from root fs (`initrd.img` is on /boot).

Usually "boot fs" doesn't have to be "root fs" (but I'm biased here as a co-author of our root-on-zfs docs). That's the case with scheme "boot pool + root pool".

So, this note still seems legit for me.

Reply:
Your proposed wording says that "Booting off zstd-compressed root pools via GRUB is not yet supported." – this is false. If it said "Booting from zstd-compressed boot pools via GRUB is not yet supported." – that would be technically true, but only relevant for BIOS boot systems with GRUB in the boot pool (and not many others), I guess. Plus, this is already said at the very top of this page, through an exhaustive listing of GRUB's supported features *and* a way to limit those thereto where it might be relevant.

Plus, it's not supported by… GRUB – the GNU project. Not OpenZFS, which is mostly, what, the FreeBSD Foundation and the U.S. government nowadays, right? (Indeed, aren't the CDDL and GPL-3+ w/ assignment wildly incompatible, making straight-forward code transplantation hilarious to consider (especially if some of the people who wrote it don't sign off on theft of their work by the FSF, which I cannot *imagine* a reason for them not to)? Do you fore-see a world within which GRUB grows *zstd* (dual BSD/GPL-2) of all things?) Do you intend to carry this note about Some Boot-loader for Some Platform being deficient into perpetuity purely because OpenSolaris/Illumos booted off it once and the FSF is a blight?

I have also written ZFS-on-root documentation – I specced sd-boot instead of GRUB. In the up-coming update for the bullseye life-cycle I will also include a mention of klapki (purely because it's easier to configure and doesn't come from under the systemd umbrella) and a few sentences on working around GRUB for platforms stuck therewith (the IBM PC).

Reply:
All I wanted to say that GRUB is still the most popular choice.

On topic - we don't note about grub on, at least, `large_dnode` in this page, so removing these parts looks consistent in this case.

Reply:
I am booting via 1GiB UEFI partition. There is the initrd.img+linux kernel und this boots up my device-mapper encrypted ZFS rootfs - with all nice options/features wanted.
I think this is the easiest way to boot current hardware today.

Just my two cents.